libyul/optimiser/CommonSwitchCasePrefixMover.cpp

```diff
@@ -0,0 +1,161 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libyul/optimiser/CommonSwitchCasePrefixMover.h>
+
+#include <libyul/optimiser/ASTCopier.h>
+#include <libyul/optimiser/NameCollector.h>
+#include <libyul/optimiser/SyntacticalEquality.h>
+#include <libyul/AST.h>
+#include <libsolutil/CommonData.h>
+
+#include <range/v3/algorithm/all_of.hpp>
+#include <range/v3/algorithm/transform.hpp>
+#include <range/v3/iterator.hpp>
+#include <range/v3/view/drop.hpp>
+#include <range/v3/view/transform.hpp>
+#include <range/v3/view/zip.hpp>
+#include <range/v3/action/transform.hpp>
+
+using namespace std;
+using namespace solidity;
+using namespace solidity::yul;
+
+namespace
+{
+class IdentifierReplacer: public ASTCopier
+{
+public:
+	IdentifierReplacer(map<YulString, YulString> _identifierMap): m_identifierMap(move(_identifierMap)) {}
+	using ASTCopier::operator();
+protected:
+	YulString translateIdentifier(YulString _name) override
+	{
+		if (YulString const* value = util::valueOrNullptr(m_identifierMap, _name))
+			return *value;
+		else
+			return _name;
+	}
```

Comment:
Note to self: This method returns a replacement for `_name` if one exists or returns identity.